src/java_bytecode/java_entry_point.cpp

```diff
@@ -589,14 +589,17 @@ bool java_entry_point(
     call_main.lhs()=return_symbol.symbol_expr();
   }
 
-  // add the exceptional return value
-  auxiliary_symbolt exc_symbol;
-  exc_symbol.mode=ID_C;
-  exc_symbol.is_static_lifetime=false;
-  exc_symbol.name=id2string(symbol.name)+EXC_SUFFIX;
-  exc_symbol.base_name=id2string(symbol.name)+EXC_SUFFIX;
-  exc_symbol.type=typet(ID_pointer, empty_typet());
-  symbol_table.add(exc_symbol);
+  if(!symbol_table.has_symbol(id2string(symbol.name)+EXC_SUFFIX))
```

Comment:
This generates a different symbol name than the catch handler in `java_bytecode_convert_method.cpp` if the main class does not have the same name as the file, e.g.:

```java
class Ex extends Throwable {}

class test2 {
  public static void main() {
   try {
     throw new Ex();
   }
   catch(Ex ex) { }
  }
}

public class test {
 public static void main (String arg[]) {
 }
}
```

When calling CBMC with --function test2.main, this will produce `"java::test2.main:()V#exception_value"` and `"java::test.main:()V#exception_value"` in `java_bytecode_convert_method.cpp`. This could probably be solved by using the class name in `java_bytecode_convert_method.cpp`.